http/responseemitter.go

```diff
@@ -212,8 +227,10 @@ func (re *responseEmitter) doPreamble(value interface{}) {
 		} else {
 			status = http.StatusInternalServerError
 		}
+		h.Set(StreamErrHeader, err.Message)
```

Comment:
This will set a *header* but not a *trailer*.

Reply:
That is, it'll only work if we haven't sent a value.

Reply:
Oh, apparently I deleted actually setting the trailer in `closeWithError`. I added it again, and now I always set it on non-nil errors.
Should I change that to only set the headers when we send a byte stream/`io.Reader`?